src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1344 @@ static ColliderCylinderInit D_80B37A5C = {
+
+    Animation_Change(&this->skelAnime, &gWolfosRunningAnim, 1.0f, 0.0f, lastFrame, ANIMMODE_LOOP_INTERP, -4.0f);
+
+    if (Rand_ZeroOne() > 0.5f) {
+        this->runAngle = 16000;
+    } else {
+        this->runAngle = -16000;
+    }
+
+    this->skelAnime.playSpeed = this->actor.speedXZ = 6.0f;
+    this->skelAnime.playSpeed *= 0.175f;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actionTimer = (Rand_ZeroOne() * 30.0f) + 30.0f;
+    this->action = WOLFOS_ACTION_RUN_AROUND_PLAYER;
+    this->runSpeed = 0.0f;
+
+    EnWf_SetupAction(this, EnWf_RunAroundPlayer);
+}
+
+void EnWf_RunAroundPlayer(EnWf* this, GlobalContext* globalCtx) {
+    s16 angle1;
+    s16 angle2;
+    s32 pad;
+    f32 distAdd = 0.0f;
+    s32 animCurFrame;
+    s32 animFrameSpeedDiff;
+    s32 animSpeed;
+    Player* player = GET_PLAYER(globalCtx);
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer + this->runAngle, 1, 4000, 1);
+
+    if (!(EnWf_DodgeRanged(globalCtx, this)) && !(EnWf_ChangeAction(globalCtx, this, 0))) {
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        angle1 = player->actor.shape.rot.y + this->runAngle + 0x8000;
+
+        if ((this->actor.bgCheckFlags & 8) ||
+            (Actor_TestFloorInDirection(&this->actor, globalCtx, this->actor.speedXZ, this->actor.shape.rot.y) == 0)) {
+            angle2 = (this->actor.bgCheckFlags & 8)
+                         ? (this->actor.wallYaw - this->actor.yawTowardsPlayer) - this->runAngle
+                         : 0;
+
+            if (ABS(angle2) > 0x2EE0) {
+                this->runAngle = -this->runAngle;
+            }
+        }
+
+        if (Actor_OtherIsTargeted(globalCtx, &this->actor)) {
+            distAdd = 150.0f;
+        }
+
+        if (this->actor.xzDistToPlayer <= (60.0f + distAdd)) {
+            Math_SmoothStepToF(&this->runSpeed, -4.0f, 1.0f, 1.5f, 0.0f);
+        } else if ((80.0f + distAdd) < this->actor.xzDistToPlayer) {
+            Math_SmoothStepToF(&this->runSpeed, 4.0f, 1.0f, 1.5f, 0.0f);
+        } else {
+            Math_SmoothStepToF(&this->runSpeed, 0.0f, 1.0f, 6.65f, 0.0f);
+        }
+
+        if (this->runSpeed != 0.0f) {
+            this->actor.world.pos.x += Math_SinS(this->actor.shape.rot.y) * this->runSpeed;
+            this->actor.world.pos.z += Math_CosS(this->actor.shape.rot.y) * this->runSpeed;
+        }
+
+        if (ABS(this->runSpeed) < ABS(this->actor.speedXZ)) {
+            this->skelAnime.playSpeed = this->actor.speedXZ * 0.175f;
+        } else {
+            this->skelAnime.playSpeed = this->runSpeed * 0.175f;
+        }
+
+        this->skelAnime.playSpeed = CLAMP(this->skelAnime.playSpeed, -3.0f, 3.0f);
+        animCurFrame = this->skelAnime.curFrame;
+        SkelAnime_Update(&this->skelAnime);
+        animFrameSpeedDiff = this->skelAnime.curFrame - ABS(this->skelAnime.playSpeed);
+        animSpeed = (f32)ABS(this->skelAnime.playSpeed);
```

Comment:
is the cast needed?

Reply:
Yes, removing it does quite a lot of damage further down.